components/feature/tabs/src/main/java/mozilla/components/feature/tabs/toolbar/TabCounterToolbarButton.kt

```diff
@@ -4,67 +4,84 @@
 
 package mozilla.components.feature.tabs.toolbar
 
-import android.view.HapticFeedbackConstants
 import android.view.View
 import android.view.ViewGroup
-import mozilla.components.browser.session.Session
-import mozilla.components.browser.session.SessionManager
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.LifecycleOwner
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.map
+import mozilla.components.browser.state.selector.getNormalOrPrivateTabs
+import mozilla.components.browser.state.selector.normalTabs
+import mozilla.components.browser.state.store.BrowserStore
 import mozilla.components.concept.toolbar.Toolbar
-import mozilla.components.feature.tabs.R
+import mozilla.components.lib.state.ext.flowScoped
 import mozilla.components.support.ktx.android.content.res.resolveAttribute
+import mozilla.components.support.ktx.kotlinx.coroutines.flow.ifChanged
 import mozilla.components.ui.tabcounter.TabCounter
+import mozilla.components.ui.tabcounter.TabCounterMenu
 import java.lang.ref.WeakReference
 
 /**
  * A [Toolbar.Action] implementation that shows a [TabCounter].
  */
+@ExperimentalCoroutinesApi
+@Suppress("LongParameterList")
 class TabCounterToolbarButton(
-    private val sessionManager: SessionManager,
-    private val showTabs: () -> Unit
+    private val lifecycleOwner: LifecycleOwner,
+    private val isPrivate: Boolean,
```

Comment:
@eliserichards we just fixed a bug in Fenix that was caused by us passing in `isPrivate` when the button gets created as opposed to looking it up dynamically. Let's remove the `isPrivate` parameter and lookup the correct tab in the flow below. Here's the change in Fenix: https://github.com/mozilla-mobile/fenix/pull/16604/commits/1e23acb8473394165581703893e79529539dc3fd

Reply:
Can we bring this change in as well so we don't regress when we're switching over to this new version in A-C?